### PR TITLE
test: fix RHEL 9.2 and 9.8 edge test compatibility

### PR DESCRIPTION
**Pull request by knecasov**
### What
Fix edge tests for `osbuild-composer` compatibility across RHEL 9.x versions.

### Why
- RHEL 9.2 `osbuild-composer` 76.1 lacks `edge-vsphere` image type, `rhel-810` distribution identifier, and `minsize` blueprint field
- RHEL 9.8 edge images use GRUB2 for bootloader installation, not `bootupd`; `bootupctl` assertion was incorrect
- `greenboot-rs` >= 0.16.0 may not ship `greenboot-success.target` preset, causing unnecessary retry loops
- RHEL 9.8 added a FIPS menuentry to `grub.cfg`, shifting boot menu positions

Assisted-by: Claude (claude-opus-4-6)

### CODE REVIEW

File: check-ostree.yaml
Comment (mcattamoredhat) [inline check-ostree.yaml]: `result_uefi` was previously consumed by `expected_boot_component` (`{{ 'Component EFI' if result_uefi.stat.exists else 'Component BIOS' }}`), which this commit deletes. 
Harmless, but it's now dead code.

Assisted-by: Claude (claude-sonnet-5)

**Review by mcattamoredhat — DISMISSED** [review]
LGTM, just a small nitpick.

File: check-ostree.yaml [hunk 1]
```diff
@@ -1214,6 +1237,29 @@
         - result_rollback is succeeded
         - greenboot_version is version('0.16.0', '>=')
 
+    # case: check greenboot-success.target is enabled (preset) after rollback
```
Comment (say-paul) [inline check-ostree.yaml]: we can check if all the greenboot services are enabled even before triggering the rollback scenarion as greenboot will fail if they  are not enabled.

Comment (knecasov) [reply check-ostree.yaml]: Do you suggest to delete this case and let only the case `check greenboot-success.target is enabled (preset)` or do you recommend editing the case `a list of greenboot* service should be enabled`, please?

Comment (say-paul) [reply check-ostree.yaml]: My suggestion is about listing all the services that needs to be enabled for greenboot > 0.16.0 : greenboot-success.target , greenboot-healthcheck.service

Comment (knecasov) [reply check-ostree.yaml]: I incorporated your suggestion. Please, let me know if it matches your idea.

**Comment by say-paul** [comment]
LGTM, a nitpick on the commit history, 251670c22bc79aa738917a96d9c7d6b64bb5cade is overridden by 2978a58504c62a339f9902dddf84b6c66bdb2fa0 so these can be merged into one. But it does not hold back the approval.

**Review by say-paul — APPROVED** [review]

File: ostree.sh [hunk 2]
```diff
@@ -202,9 +202,10 @@ EOF
 sudo sed -i 's/default="1"/default="3"/' "${GRUB_CFG}"
 sudo sed -i 's/timeout=60/timeout=10/' "${GRUB_CFG}"
 
-# For CentOS Stream test, grub default menuentry has to be changed to workaround issue
+# For CentOS Stream and RHEL 9.8+, grub default menuentry has to be changed
+# because a FIPS menuentry was added, shifting the position of our custom entry.
 # https://github.com/virt-s1/rhel-edge/issues/9968
-if [[ "${ID}-${VERSION_ID}" == "centos-9" ]]; then
+if [[ "${ID}-${VERSION_ID}" == "centos-9" ]] || [[ "${ID}" == "rhel" && "$(echo "${VERSION_ID}" | tr -d '.')" -ge 98 ]]; then
```
Comment (mcattamoredhat) [inline ostree.sh]: `VERSION_ID` for RHEL 8.10 is `"8.10"`. Stripping the dot gives `"810"`, and `810 -ge 98` is true. This condition was intended to match only RHEL ≥9.8, but it will also match `rhel-8.10`, which `ostree.sh` explicitly supports (line 51 case statement) and which is actively tested via `/test-rhel-8-10` (the `edge-x86-commit` FMF plan has no `rhel-8-10` exclusion). Unless RHEL 8.10 also happens to have a shifted FIPS menu entry (not stated anywhere in the PR), this will silently point GRUB's default boot entry to entry 4 instead of 3 on RHEL 8.10, which is very likely to break that test path.

Suggested simple check:

```
RHEL_MAJOR="${VERSION_ID%%.*}"
RHEL_MINOR="${VERSION_ID#*.}"
if [[ "${ID}-${VERSION_ID}" == "centos-9" ]] || [[ "${ID}" == "rhel" && "${RHEL_MAJOR}" == "9" && "${RHEL_MINOR}" -ge 8 ]]; then
    sudo sed -i 's/default="3"/default="4"/' "${GRUB_CFG}"
fi
```

**Comment by knecasov** [comment]
> LGTM, a nitpick on the commit history, [251670c](https://github.com/virt-s1/rhel-edge/commit/251670c22bc79aa738917a96d9c7d6b64bb5cade) is overridden by [2978a58](https://github.com/virt-s1/rhel-edge/commit/2978a58504c62a339f9902dddf84b6c66bdb2fa0) so these can be merged into one. But it does not hold back the approval.

Thank you very much for pointing this out. I incorporated your note.

File: ostree.sh (hunk 2 above)
Comment (knecasov) [reply ostree.sh]: Thank you very much for catching it! I incorporated your suggestion.

**Review by say-paul — APPROVED** [review]

**Comment by knecasov** [comment]
An explanation why some tests fail.

RHEL 9.2
- `edge-x86-9to9` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
- `edge-x86-ami-image` - a failure due to authentication issues
- `edge-x86-installer` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side

RHEL 9.8
- `edge-x86-8to9` - an error tracked in issue #11498
- `edge-x86-9to9` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
- `edge-x86-ami-image` - a failure due to authentication issues
- `edge-x86-commit` - an error tracked in issue #12302
- `edge-x86-installer` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
- `edge-x86-vsphere` - a failure due to authentication issues

**Review by djach7 — APPROVED** [review]
LGTM — verified the version parsing fix handles edge cases like 8.10 correctly, greenboot gating logic is sound, and FMF exclusions match the script changes

**Comment by mcattamoredhat** [comment]
> An explanation why some tests fail.
> 
> RHEL 9.2
> 
> * `edge-x86-9to9` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
> * `edge-x86-ami-image` - a failure due to authentication issues
> * `edge-x86-installer` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
> 
> RHEL 9.8
> 
> * `edge-x86-8to9` - an error tracked in issue [rhel-9.6 edge-x86-8to9 test is failing due to kdump #11498](https://github.com/virt-s1/rhel-edge/issues/11498)
> * `edge-x86-9to9` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
> * `edge-x86-ami-image` - a failure due to authentication issues
> * `edge-x86-commit` - an error tracked in issue [Anaconda failing to install bootloader appears again in RHEL-9.8 edge-commit test case #12302](https://github.com/virt-s1/rhel-edge/issues/12302)
> * `edge-x86-installer` - a maximum test time exceeded, it seems to be a problem on a Testing Farm side
> * `edge-x86-vsphere` - a failure due to authentication issues

* `edge-x86-9to9` - test is actually ok, failure is due to testing farm known issue.

**Review by mcattamoredhat — APPROVED** [review]
LGTM. Failures are not related to the changes of this PR, and need to be addressed in separated PRs.
